### ISSUE TITLE
Image views, when used-after-free in a framebuffer, does not raise validation error; segfaults under renderdoc.

### ISSUE THREAD

**Issue by cwfitzgerald** [KhronosGroup/Vulkan-ValidationLayers#10448]
**Describe the situation in which you encountered the missing validation**

Get setup up:

```bash
git clone https://github.com/gfx-rs/wgpu.git
git checkout v26.0.1
cargo build -p wgpu-examples
```

This will build a binary in `target/debug/wgpu-examples.exe`. (remove exe for *nix). Run this binary with the argument `cube`. If you are running on an AMD gpu, you will notice that the rendering may flicker. This is because of the code using a framebuffer which points to a deleted image view. The program automatically enables the validation layers, if they are available. There are no validation warnings, however. You can validate this by setting the env var `RUST_LOG=wgpu_hal=info` to see info level logs.

On any gpu, run the program under renderdoc. The program will crash. If you run under renderdoc and a debugger, you will notice it crashes inside the validation layers, while trying to validate the image views.

This was a bit convoluted to find, so let me know if anything else is needed.

**Valid Usage IDs requested**
I'm honestly unsure where to find the rules against use-after-free.

**Additional Context**
Windows 11
Replicated on Nvidia, AMD.

**Comment by spencer-lunarg** [comment]
So this all falls under the Object Tracker stuff

For example we should be catching things like this

<img width="454" height="66" alt="Image" src="https://github.com/user-attachments/assets/e6740248-3289-4057-a70c-1e24c5f4ce3f" />

and then at draw calls, we have ton of logic that will mark the command buffer as "invalid" if something inside of it was deleted (like a pipeline)... the issue is there are enough indirection of objects relying on objects that this example might be missed

@ziga-lunarg might beat me to writing a test, but will try and emulate in a test tomorrow

**Comment by ziga-lunarg** [comment]
The simple tests I have does report an error:

> Validation Error: [ VUID-vkQueueSubmit-pCommandBuffers-00070 ] | MessageID = 0xaeecdc0b
> vkQueueSubmit(): pSubmits[0].pCommandBuffers[0] was called in VkCommandBuffer 0x2d8d8b54510 which is invalid because the bound VkImageView 0x60000000006 was destroyed.
> The Vulkan spec states: Each element of the pCommandBuffers member of each element of pSubmits must be in the pending or executable state (https://docs.vulkan.org/spec/latest/chapters/cmdbuffers.html#VUID-vkQueueSubmit-pCommandBuffers-00070)
> Objects: 3
>     [0] VkImageView 0x60000000006
>     [1] VkFramebuffer 0x80000000008
>     [2] VkCommandBuffer 0x2d8d8b54510

```
TEST_F(NegativeRenderPass, DestroyedImageViews) {
    RETURN_IF_SKIP(Init());

    vkt::Image image(*m_device, 32u, 32u, VK_FORMAT_R8G8B8A8_UNORM, VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT);
    vkt::ImageView image_view = image.CreateView(VK_IMAGE_ASPECT_COLOR_BIT);

    VkAttachmentDescription attachment = {};
    attachment.format = VK_FORMAT_R8G8B8A8_UNORM;
    attachment.samples = VK_SAMPLE_COUNT_1_BIT;
    attachment.loadOp = VK_ATTACHMENT_LOAD_OP_CLEAR;
    attachment.storeOp = VK_ATTACHMENT_STORE_OP_STORE;
    attachment.initialLayout = VK_IMAGE_LAYOUT_UNDEFINED;
    attachment.finalLayout = VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;

    VkAttachmentReference color_attachment;
    color_attachment.attachment = 0u;
    color_attachment.layout = VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;

    VkSubpassDescription subpass = {};
    subpass.pipelineBindPoint = VK_PIPELINE_BIND_POINT_GRAPHICS;
    subpass.colorAttachmentCount = 1u;
    subpass.pColorAttachments = &color_attachment;

    VkRenderPassCreateInfo rp_ci = vku::InitStructHelper();
    rp_ci.attachmentCount = 1u;
    rp_ci.pAttachments = &attachment;
    rp_ci.subpassCount = 1u;
    rp_ci.pSubpasses = &subpass;
    vkt::RenderPass rp(*m_device, rp_ci);
    vkt::Framebuffer fb(*m_device, rp, 1u, &image_view.handle(), 32u, 32u);

    CreatePipelineHelper pipe(*this);
    pipe.gp_ci_.renderPass = rp;
    pipe.CreateGraphicsPipeline();

    VkClearValue clear_value;
    clear_value.color.float32[0] = 0.0f;
    clear_value.color.float32[1] = 0.0f;
    clear_value.color.float32[2] = 0.0f;
    clear_value.color.float32[3] = 0.0f;

    VkRenderPassBeginInfo rp_bi = vku::InitStructHelper();
    rp_bi.renderPass = rp;
    rp_bi.framebuffer = fb;
    rp_bi.renderArea = {{0, 0}, {32u, 32u}};
    rp_bi.clearValueCount = 1u;
    rp_bi.pClearValues = &clear_value;

    m_command_buffer.Begin();
    m_command_buffer.BeginRenderPass(rp_bi);
    vk::CmdBindPipeline(m_command_buffer, VK_PIPELINE_BIND_POINT_GRAPHICS, pipe);
    vk::CmdDraw(m_command_buffer, 4u, 1u, 0u, 0u);
    m_command_buffer.EndRenderPass();
    m_command_buffer.End();

    image_view.Destroy();

    m_default_queue->SubmitAndWait(m_command_buffer);
}
```

@cwfitzgerald At what point is the image view destroyed?

**Comment by ziga-lunarg** [comment]
Also slightly related #9239

**Comment by spencer-lunarg** [comment]
@cwfitzgerald update, I grabbed wgpu and tried on my NVIDIA laptop (with AMD cpu integrated graphics)

I see the flickering on AMD, but when I use validation, the flickering goes away and there is no validation errors... looking into now more

**Comment by spencer-lunarg** [comment]
@ziga-lunarg from an API dump, the sequence looks like

```
vkEndCommandBuffer
vkQueueSubmit
vkGetSemaphoreCounterValue
vkResetCommandPool
vkDestroyImageView (framebuffer image)
vkDestroyImageView (clear surface texture view)
vkQueuePresentKHR
vkWaitSemaphores
vkAcquireNextImageKHR
```

**Comment by cwfitzgerald** [comment]
@spencer-lunarg Yup, that sounds right - thanks for pulling the api trace!
